astropy/cosmology/flrw/base.py

```diff
@@ -87,25 +106,25 @@ def scale_factor0(self):
         the redshift of interest for that paper. Explicitly using :math:`a_0` in both
         calculation and code avoids ambiguity.
         """
-        return u.Quantity(self.scale_factor(0), unit=u.one)
+        return self.scale_factor(0) << u.one
 
     @deprecated_keywords("z", since="7.0")
-    def scale_factor(self, z):
+    def scale_factor(self, z) -> Quantity | NDArray[Any] | float:
         """Scale factor at redshift ``z``.
 
         The scale factor is defined as :math:`a = 1 / (1 + z)`.
 
         Parameters
         ----------
-        z : Quantity-like ['redshift'], array-like
+        z : Quantity-like ['redshift'], array-like, positional-only
```

Comment:
Do you consider this an oversight from #16597 ? My understanding was that it was best to avoid this annotation over the deprecation period. If you disagree, I think it should be changed consistently.

Reply:
Hm. Yes. If the method raises a warning when passed by key word, it's probably best to tell users it's positional-only.
Let's change them all in a followup PR.